HW17/Vector2d.cpp

```diff
@@ -0,0 +1,75 @@
+#include "Vector2d.h"
+#include <cmath>
+#include <iostream>
+
+int Vector2d::instanceCount = 0;
+
+Vector2d::Vector2d()
+    :x(0.0f), y(0.0f)
+{    ++instanceCount; }
+
+Vector2d& Vector2d::operator=(const Vector2d& other)
+{
+    if(this != &other){
+        x = other.x;
+        y = other.y;
+    }
+    return *this;
+}
+
+Vector2d::Vector2d(float xVal, float yVal)
+    : x(xVal), y(yVal)
+{  ++instanceCount; }
+
+float Vector2d::getX() const
+{
+    return x;
+}
+
+float Vector2d::getY() const
+{
+    return y;
+}
```

Comment:
it's ok to put one-liner in header.
In most cases they can even be `inlined` automatically for performance optimization
https://www.geeksforgeeks.org/inline-functions-cpp/